### DISCUSSION TITLE
Any way to simulate a suffix in container names?

### DISCUSSION THREAD

**Discussion by sneko** [testcontainers/testcontainers-node#415] [Q&A] [answered]
Hi @cristianrgreco ,

I'm parsing a `docker-compose.yaml` thanks to `DockerComposeEnvironment(...)` but unfortunately sometimes I'm already running this docker-compose locally for developing purpose. So when my Jest tests run, they fail with:
```
Conflict. The container name "/xxxxxxxx" is already in use by container
```

What do you advise so that when using your library with a docker-compose file it sets a random suffix (or at least a suffix) so it does not conflict with a local `docker-compose up`? Something programatic?

Otherwise I thought about setting an environment variable in the `docker-compose.yaml` as a suffix for each `container_name`, so when nothing defined, it's the root name, and inside a Jest test I would set `process.env.DOCKER_COMPOSE_CONTAINERS_SUFFIX = "mytests"` before using your library (but not sure it will replace things at that time).

Thank you for sharing your thoughts on this :)

**Comment by cristianrgreco** [comment]
Hi @sneko, no there isn't a way. This is just how docker-compose works, if you tried to start the compose file again Docker would give you the same error. I don't think testcontainers should behave differently, I think if it did, i.e connect to an existing container else start one, it would lead to unexpected results for most people.

I'd recommend you remove `container_name` from your docker-compose files as this is what causes the conflict. I don't see any need for it, if your containers need to talk to each other then you can add network aliases instead.

**Reply by sneko** [reply]
@cristianrgreco , in my case the `container_name` helps me to get the `host` + `mappedPort` once up (so my tests running outside the services from the docker-compose are able to reach the database):
https://github.com/inclusion-numerique/mediature/blob/dev/apps/main/utils/database.ts#L34

But indeed, maybe there is a way to have a fixed `host+mappedPort` reachable from the outside, but not sure yet how to do that hmmm

**Reply by sneko** [reply, marked as answer]
as mentioned in my initial question, I adopted the solution of suffixing my container names, so in my `docker-compose.yaml` I get for example:
```
version: '3.5'

services:
  postgres:
    container_name: postgres_container_${DOCKER_COMPOSE_CONTAINER_NAME_SUFFIX:-default}
```

Then in my test Jest environnement I do:
```ts
  const environment = await new DockerComposeEnvironment(composeFilePath, composeFile)
    .withEnvironment({
      DOCKER_COMPOSE_CONTAINER_NAME_SUFFIX: 'jest',
      POSTGRES_USER: dummyUser,
      POSTGRES_PASSWORD: dummyPassword,
      POSTGRES_DB: dummyDatabase,
    })
```

And to get the container so I can do `.getMappedPort` I simply use:
```ts
  const container = environment.getContainer('postgres_container_jest');
```

Like that I have no conflict on my computer when running a "docker-compose up" for local development, and Jest that run in parallel.

Hope it helps,
